pkg/gateway/api/metrics_event.go

```diff
@@ -0,0 +1,513 @@
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
+	if ev.Labels != nil {
+		tag = ev.Labels["tag"]
+	}
+	sdkErrorCounter.WithLabelValues(
+		env,
+		tag,
+		errorType,
+		ev.ApiId.String(),
+		event.SdkVersion,
+		event.SourceId.String(),
+	).Inc()
+	return nil
+}
+
+func (s *grpcGatewayService) saveTimeoutError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeTimeout
+	ev := &eventproto.TimeoutErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
+	if ev.Labels != nil {
+		tag = ev.Labels["tag"]
+	}
+	sdkErrorCounter.WithLabelValues(
+		env,
+		tag,
+		errorType,
+		ev.ApiId.String(),
+		event.SdkVersion,
+		event.SourceId.String(),
+	).Inc()
+	return nil
+}
+
+func (s *grpcGatewayService) saveInternalError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeInternal
+	ev := &eventproto.InternalErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
+	if ev.Labels != nil {
+		tag = ev.Labels["tag"]
+	}
+	sdkErrorCounter.WithLabelValues(
+		env,
+		tag,
+		errorType,
+		ev.ApiId.String(),
+		event.SdkVersion,
+		event.SourceId.String(),
+	).Inc()
+	return nil
+}
+
+func (s *grpcGatewayService) saveNetworkError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeNetwork
+	ev := &eventproto.NetworkErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
+	if ev.Labels != nil {
+		tag = ev.Labels["tag"]
+	}
+	sdkErrorCounter.WithLabelValues(
+		env,
+		tag,
+		errorType,
+		ev.ApiId.String(),
+		event.SdkVersion,
+		event.SourceId.String(),
+	).Inc()
+	return nil
+}
+
+func (s *grpcGatewayService) saveInternalSdkError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeSDKInternal
+	ev := &eventproto.InternalSdkErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
+	if ev.Labels != nil {
+		tag = ev.Labels["tag"]
+	}
+	sdkErrorCounter.WithLabelValues(
+		env,
+		tag,
+		errorType,
+		ev.ApiId.String(),
+		event.SdkVersion,
+		event.SourceId.String(),
+	).Inc()
+	return nil
+}
+
+func (s *grpcGatewayService) saveUnknownError(event *eventproto.MetricsEvent, env string) error {
+	errorType := ErrorTypeUnknown
+	ev := &eventproto.UnknownErrorMetricsEvent{}
+	if err := ptypes.UnmarshalAny(event.Event, ev); err != nil {
+		return err
+	}
+	if ev.ApiId == eventproto.ApiId_UNKNOWN_API {
+		return MetricsSaveErrUnknownApiId
+	}
+	var tag string
```

Comment:
Do we need to insert "none" if tag is empty?
https://github.com/bucketeer-io/bucketeer/blob/main/pkg/eventpersisterdwh/persister/evaluation_event.go#L166

Reply:
Not anymore. I have confirmed that all projects are using the latest SDK.